scripts/tests/calibnet_stateless_mode_check.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -euxo pipefail
+
+# This script tests the stateless mode of a forest node
+
+source "$(dirname "$0")/harness.sh"
+
+forest_init_stateless
+
+echo "Verifying the heaviest tipset to be the genesis"
+MSG=$($FOREST_CLI_PATH chain head)
+assert_eq "$MSG" $'[\n  "bafy2bzacecyaggy24wol5ruvs6qm73gjibs2l2iyhcqmvi7r7a4ph7zx3yqd4"\n]'
+
+wait_util_p2p_is_ready
+ADDRESS=$($FOREST_CLI_PATH net listen | tail -n 1)
+echo "Stateless node address: $ADDRESS"
+PEER_ID=$(echo "$ADDRESS" | cut -d "/" -f7 -z)
+echo "Stateless node peer id: $PEER_ID"
+
+# Run a normal forest node that only connects to the stateless node
+CONFIG_PATH="./forest_config.toml"
+{
+    echo "[network]"
+    echo "listening_multiaddrs = [\"/ip4/127.0.0.1/tcp/0\"]"
+    echo "bootstrap_peers = [\"$ADDRESS\"]"
+} > $CONFIG_PATH
+
+NODE_LOG_DIRECTORY=$(mktemp --directory)
+RUST_LOG="info,forest_filecoin::chain_sync::network_context=debug" $FOREST_PATH --chain calibnet --encrypt-keystore false --auto-download-snapshot --config "$CONFIG_PATH" --no-metrics --rpc false --log-dir "$NODE_LOG_DIRECTORY" &
+PID=$!
+# Verify that the stateless node can respond to chain exchange requests
```

Comment:
I think a timeout would help here. I can imagine the expression not matching because someone changed one letter. Then, these two lines would wait until killed by GH.

Reply:
It's also a bit flaky to rely on logs. Maybe we could rely on metrics instead?

Reply:
We don't have such chain exchange metrics for each peer id. Could you elaborate?

Reply:
> Add 10min timeout to the workflow step.

The problem with the workflow timeout is that it won't tell you precisely what was sleeping for so long, and it won't timeout on a local test.

Reply:
> We don't have such chain exchange metrics for each peer id. Could you elaborate?

We could add some. We'd need to be tactical to not [bloat](https://github.com/ChainSafe/forest/issues/3509) the metric reports. We could have a metric of, e.g., chain exchanges per peer in the last 5 minutes.

Reply:
@LesnyRumcajs How about adding per-peer chain exchange metrics after https://github.com/ChainSafe/forest/issues/3509 is tackled? For now, it will bloat the metrics report just like `peer_tipset_epoch`. Or would you suggest bloat it first then make the fix later together with https://github.com/ChainSafe/forest/issues/3509

Reply:
I suggest not bloating it. I imagine metrics with labels and a timed cache of the most recent peers that did the chain exchange. Periodically, the metrics that expired would be deleted with something like https://docs.rs/prometheus/latest/prometheus/core/struct.MetricVec.html#method.remove_label_values, and so the `/metrics` page would not contain those peers.

It doesn't touch #3509 (though it'd potentially introduce something that could be used there).

If you believe implementing it would be a lot of work, we can do that in a separate PR. What's your pick?

Reply:
@LesnyRumcajs Fixed.

Reply:
Beautiful!